### PR TITLE
BUG: Set image origin according to point set bounding box

### PR DESCRIPTION
**Pull request by jhlegarreta**
- ENH: Add `itk::PointSetToImageFilter` unit tests
- BUG: Set image origin according to point set bounding box

## PR Checklist
- [X] No [API changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#breaking-changes) were made (or the changes have been approved)
- [X] No [major design changes](https://github.com/InsightSoftwareConsortium/ITK/blob/master/CONTRIBUTING.md#design-changes) were made (or the changes have been approved)
- [X] Added test (or behavior not changed)
- [X] Added [license](https://github.com/InsightSoftwareConsortium/ITK/blob/master/Utilities/KWStyle/ITKHeader.h) to new files (if any)

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Supersedes PR #3280 (deleted branch by mistake).

File: Modules/Core/Common/test/itkPointSetToImageFilterTest1.cxx [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+  unsigned int count = 0;
+  for (float theta = 0; theta < 2.0 * itk::Math::pi; theta += 0.1)
+  {
+    PointType point;
+    point[0] = radius * std::cos(theta);
+    point[1] = radius * std::sin(theta);
+
+    pointSet->SetPoint(count, point);
+    count++;
+  }
```
Comment (jhlegarreta) [inline Modules/Core/Common/test/itkPointSetToImageFilterTest1.cxx:62]: This was borrowed from other examples/tests, e.g.https://github.com/InsightSoftwareConsortium/ITK/blob/d715b098ba992404af994d7d009a066c42f8c7f7/Modules/Registration/Metricsv4/test/itkExpectationBasedPointSetMetricRegistrationTest.cxx

For future uses, best would be to relocate such parts to a testing utils module where they can be re-used across the code base. It is left for a separate PR.

**Comment by jhlegarreta** [comment]
Just for the records, the vascular point set was taken from https://github.com/InsightSoftwareConsortium/ITK/blob/017f7f501978830e22ebba98831b258b55371c08/Examples/Data/IterativeClosestPointFixedPoints2.txt

File: Modules/Core/Common/include/itkPointSetToImageFilter.hxx [hunk 2]
```diff
@@ -130,7 +130,7 @@ PointSetToImageFilter<TInputPointSet, TOutputImage>::GenerateData()
   for (i = 0; i < InputPointSetDimension; ++i)
   {
     size[i] = static_cast<SizeValueType>(bb->GetBounds()[2 * i + 1] - bb->GetBounds()[2 * i]);
-    origin[i] = 0; // bb->GetBounds()[2*i];
+    origin[i] = bb->GetBounds()[2 * i];
   }
```
Comment (jhlegarreta) [inline Modules/Core/Common/include/itkPointSetToImageFilter.hxx:134]: Probably setting the output image's properties (origin, spacing, etc.), i.e. a few dozen lines below, should dwell in a separate e.g. protected method for an improved encapsulation. Left for a separate PR.

**Review by hjmjohnson — APPROVED** [review]

**Pull request merged**